pkg/client/client.go

```diff
@@ -145,6 +145,16 @@ func (c ApplyClient) applyObject(ctx context.Context, obj client.Object, options
 	originalGeneration := existing.GetGeneration()
 	obj.SetResourceVersion(existing.GetResourceVersion())
 
+	// Special handling of ServiceAccounts is required because if a ServiceAccount is reapplied when it already exists, it causes Kubernetes controllers to
+	// automatically create new Secrets for the ServiceAccounts. After enough time the number of Secrets created will hit the Secrets quota and then no new
+	// Secrets can be created. To prevent this from happening, we keep the existing refs to secrets.
+	if strings.EqualFold(obj.GetObjectKind().GroupVersionKind().Kind, "ServiceAccount") {
+		MergeAnnotations(obj, existing.GetAnnotations()) // copy existing annotations
+		if err := RetainSecrets(obj, existing); err != nil {
+			return false, err
```

Comment:
btw, there are also `imagePullSecrets`. 
I wouldn't make it so complicated - was there anything wrong with the original code when we just simply updated the labels & annotations in the existing one and used that object to update the resource?

Reply:
I wanted to reuse the existing Get/Create/Update calls from this function, I'm merging the labels and annotations on the existing object in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/cf4f268fefbf595507d343c3d97b36e593143235

I'm unsure about the ResourceVerision that gets overwritten, not sure if that's an issue.